pandas/tools/merge.py

```diff
@@ -1202,16 +1202,11 @@ def _sort_labels(uniques, left, right):
         # tuplesafe
         uniques = Index(uniques).values
 
-    sorter = uniques.argsort()
+    l = len(left)
+    labels = np.concatenate([left, right])
```

Comment:
xref to this: https://github.com/pydata/pandas/issues/4588

This may/may not be impacted by this issue (and that should be a separate PR to address anyhow)